test/ServerRequestFilter/FilterUsingXForwardedHeadersTest.php

```diff
@@ -368,4 +368,30 @@ public function testOnlyHonorsXForwardedProtoIfValueResolvesToHTTPS(
         $uri             = $filteredRequest->getUri();
         $this->assertSame($expectedScheme, $uri->getScheme());
     }
+
+    public function testPortExtractedFromXForwardedHostHeader(): void
+    {
+        $request = new ServerRequest(
+            ['REMOTE_ADDR' => '192.168.1.1'],
+            [],
+            'http://localhost:80/foo/bar',
+            'GET',
+            'php://temp',
+            [
+                'Host'              => 'localhost',
+                'X-Forwarded-Host'  => 'example.com:4433',
+                'X-Forwarded-Port'  => '8433',
+                'X-Forwarded-Proto' => 'https',
+            ]
+        );
+
+        $filter = FilterUsingXForwardedHeaders::trustAny();
+
+        $filteredRequest = $filter($request);
+        $filteredUri     = $filteredRequest->getUri();
+        $this->assertNotSame($request->getUri(), $filteredUri);
+        $this->assertSame('example.com', $filteredUri->getHost());
+        $this->assertSame(4433, $filteredUri->getPort());
```

Comment:
Prior to the fix, we didn't consider `X-Forwarded-Port`, nor did we consider `X-Forwarded-Host` values with ports included; in the latter case, this likely would have led to an exception thrown by the `Uri` instance during creation, as the host value would fail assertions.

Since we honor `X-Forwarded-Port` now, I think it makes sense to provide it, and it's technically a bug in even the 2.9.2 implementation.